mcs/class/System/Test/System.Net/DnsTest.cs

```diff
@@ -205,12 +206,21 @@ public void GetHostByName ()
 			}
 		}
 
+		static IPAddress GetNonIPv6Address (IPHostEntry h)
+		{
+			var al0 = h.AddressList [0];

```

Comment:
Sure, it make sense for mixed networks :) 

Note that _would still work_ is incorrect. Right now an IPv6 address, if first in list, would fail some of the tests. Either no bots are running under such network or there's something (by design or by luck) making the IPv4 address come on top of the list.